### PR TITLE
sql-statements: add info for amend txn when column type changing

### PR DESCRIPTION
**Pull request by zimulala**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/issues/26203

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-change-column.md [hunk 1]
```diff
@@ -157,6 +157,7 @@ ERROR 8200 (HY000): Unsupported modify column: change from original type decimal
 * 不支持分区表上的列类型变更。
 * 不支持生成列上的列类型变更。
 * 不支持部分数据类型（例如，部分时间类型、Bit、Set、Enum、JSON 等）的变更，因为 TiDB 中 `CAST` 函数与 MySQL 的行为存在兼容性问题。
+* 不支持在使用悲观事务时，将 [`tidb_enable_amend_pessimistic_txn`](/system-variables.md#tidb_enable_amend_pessimistic_txn-从-v407-版本开始引入) 设置成 `ON` 的场景。
```
Comment (tangenta) [inline sql-statements/sql-statement-change-column.md:160]: This seems unrelated to "MySQL Compatibility"

File: sql-statements/sql-statement-modify-column.md [hunk 2]
```diff
@@ -215,6 +215,7 @@ CREATE TABLE `t1` (
     ALTER TABLE t MODIFY COLUMN a DATETIME;
     ERROR 8200 (HY000): Unsupported modify column: change from original type decimal(13,7) to datetime is currently unsupported yet
     ```
+* 不支持在使用悲观事务时，将 [`tidb_enable_amend_pessimistic_txn`](/system-variables.md#tidb_enable_amend_pessimistic_txn-从-v407-版本开始引入) 设置成 `ON` 的场景。
```
Comment (tangenta) [inline sql-statements/sql-statement-modify-column.md:218]: Maybe we should add this description to `tidb_enable_amend_pessimistic` instead because the users are more likely to search that system variable first.

File: sql-statements/sql-statement-change-column.md (hunk 1 above)
Comment (zimulala) [reply sql-statements/sql-statement-change-column.md:160]: None of the above details is written in the compatibility section

File: sql-statements/sql-statement-modify-column.md (hunk 2 above)
Comment (AilinKid) [reply sql-statements/sql-statement-modify-column.md:218]: +1, we better copy it to under the item of `tidb_enable_amend_pessimistic_txn`

File: sql-statements/sql-statement-change-column.md (hunk 1 above)
Comment (tangenta) [reply sql-statements/sql-statement-change-column.md:160]: The title of this section is "## MySQL 兼容性"

Comment (zimulala) [reply sql-statements/sql-statement-change-column.md:160]: Got it.

**Comment by TomShawn** [comment]
@tangenta @AilinKid PTAL again. Thanks!

File: system-variables.md [hunk 3]
```diff
@@ -472,7 +472,8 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 
 > **注意：**
 >
-> 目前该特性可能造成事务语义的变化，且与 TiDB Binlog 存在部分不兼容的场景，可以参考[事务语义行为区别](https://github.com/pingcap/tidb/issues/21069)和[与 TiDB Binlog 兼容问题汇总](https://github.com/pingcap/tidb/issues/20996)了解更多关于该特性的使用注意事项。
+> - 目前该特性可能造成事务语义的变化，且与 TiDB Binlog 存在部分不兼容的场景，可以参考[事务语义行为区别](https://github.com/pingcap/tidb/issues/21069)和[与 TiDB Binlog 兼容问题汇总](https://github.com/pingcap/tidb/issues/20996)了解更多关于该特性的使用注意事项。
+> - 从 v5.1.0 版本起，TiDB 开始支持 Reorg 类型变更的 [MODIFY/CHANGE COLUMN](/sql-statements/sql-statement-modify-column.md)，但不兼容此变量功能。
```
Comment (tangenta) [inline system-variables.md:476]: I think we don't need to change this because it is mentioned in the previous paragraph:

```
- `MODIFY COLUMN` 或 `CHANGE COLUMN` 类型的 DDL 操作，且只对增大字段长度的操作生效。
```

Comment (AilinKid) [reply system-variables.md:476]: normal `MODIFY COLUMN` is compatible with `tidb_enable_amend_pessimistic_txn` except reorg type

**Comment by zimulala** [comment]
This issue is fixed(https://github.com/pingcap/tidb/pull/26269) and cherry-picked to v5.1.
